src/i965_encoder.c

```diff
@@ -808,8 +810,28 @@ intel_encoder_check_misc_parameter(VADriverContextP ctx,
         VAEncMiscParameterBufferQualityLevel* param_quality_level = (VAEncMiscParameterBufferQualityLevel*)pMiscParam->data;
         encoder_context->quality_level = param_quality_level->quality_level;
 
-        if (encoder_context->quality_level == 0)
-            encoder_context->quality_level = ENCODER_DEFAULT_QUALITY;
+        if (encoder_context->quality_level == 0) {
```

Comment:
default quality level value in the function intel_enc_hw_context_init() is set to 4. so here if(==0) will never be called. how about removing these changes?